### PR TITLE
stub: clean up client calls to avoid synthetic class

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 1]
```diff
@@ -47,7 +47,8 @@
  * that the runtime can vary behavior without requiring regeneration of the stub.
  */
 public final class ClientCalls {
-  private static final Logger log = Logger.getLogger(ClientCalls.class.getName());
+  // Non private to avoid synthetic class
+  static final Logger log = Logger.getLogger(ClientCalls.class.getName());
```
Comment (zhangkun83) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: Maybe just move this logger to the class where it is used?

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 2]
```diff
@@ -329,13 +332,15 @@ public void setMessageCompression(boolean enable) {
     }
   }
 
-  private static class StreamObserverToCallListenerAdapter<ReqT, RespT>
+  // Non private to avoid synthetic class
+  static final class StreamObserverToCallListenerAdapter<ReqT, RespT>
```
Comment (zhangkun83) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: This is only used in static methods of the outer class. I thought outer class have direct access to its private inner classes, thus only the constructor needed to be non-private. Isn't it the case?

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 3]
```diff
@@ -392,11 +397,12 @@ public void onReady() {
   /**
    * Complete a GrpcFuture using {@link StreamObserver} events.
    */
-  private static class UnaryStreamToFuture<RespT> extends ClientCall.Listener<RespT> {
+  // Non private to avoid synthetic class
+  static final class UnaryStreamToFuture<RespT> extends ClientCall.Listener<RespT> {
```
Comment (zhangkun83) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: ditto

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 4]
```diff
@@ -536,7 +544,8 @@ public void remove() {
       throw new UnsupportedOperationException();
     }
 
-    private class QueuingListener extends ClientCall.Listener<T> {
+    // Non private to avoid synthetic class
+    final class QueuingListener extends ClientCall.Listener<T> {
```
Comment (zhangkun83) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: Ditto. This is only used by its outer class.

File: stub/src/main/java/io/grpc/stub/ClientCalls.java (hunk 1 above)
Comment (carl-mastrangelo) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: Done.

File: stub/src/main/java/io/grpc/stub/ClientCalls.java (hunk 2 above)
Comment (carl-mastrangelo) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: You are right.  I have made this change.

**Comment by carl-mastrangelo** [comment]
@zhangkun83 PTAL

**Comment by carl-mastrangelo** [comment]
Jenkins retest this please

**Pull request merged**